File: content/snippets/scala2/language_basics_file_paths_and_directories.scala
import java.nio.file.{Files, Paths}

object FilePathsAndDirectoriesBasics {
  def main(args: Array[String]): Unit = {
    val filePath = Paths.get("demo_folder", "sub", "data.txt")
    val parent = filePath.getParent

    Files.createDirectories(parent)

    println(s"file path: $filePath")
    println(s"parent path: $parent")
    println(s"directory exists: ${Files.exists(parent)}")
  }
}

File: content/snippets/python/language_basics_file_paths_and_directories.py
#!/usr/bin/env python3
from pathlib import Path


def main() -> None:
    base = Path("demo_folder")
    file_path = base / "sub" / "data.txt"
    file_path.parent.mkdir(parents=True, exist_ok=True)

    print("file path:", file_path)
    print("parent path:", file_path.parent)
    print("directory exists:", file_path.parent.exists())


if __name__ == "__main__":
    main()
